### ISSUE TITLE
Accessing the contents of a non existing translation file should not raise error?

### ISSUE THREAD

**Issue by goanpeca** [lektor/lektor#459]
Currently I want to add an `Edit on Github` link to generated files. I am using alternatives and not all webpages have a real `content+<alt>.lr` page, so in these cases I want to hide the link for pages that do not really exist on github (yet).

The code I am using to check this is:

```jinja2
{% set trans_available = this.contents.as_text() != site.get(this.path).contents.as_text() %}
```

The problem is that for non existing alterinatives `this.contents` raises a `IOError: [Errno 2] No such file or directory:`

Maybe there is another way to do this but, in this case I suggest we return `None` as the contents to indicate the file does not exists (instead of raising an error) or adding an extra property, to check if the file exists  `obj.content_exists`?. I can implement either option.

Thoughts @mitsuhiko, @nixjdm ?

**Comment by rlaverde** [comment]
I think this is related to #270, falling back to the default translation could be better

**Comment by goanpeca** [comment]
No @rlaverde this is a different issue, cause I want to actively know what translations exists and which translations do not exist so I can act on it to display or not a link. I don't want to fallback on the default locale

**Comment by nixjdm** [comment]
I haven't used alternatives before, so I'm playing around in the example project. It looks like `this` is working according to the [chart](https://www.getlektor.com/docs/content/alts/#what-s-loaded), but the `this.contents` doesn't match what is in the "File Used" column. Instead, with the setup on that docs page, `this.contents` on / looks for  
```
IOError: [Errno 2] No such file or directory: u'/home/joe/workspace/lektor/example/content/contents+en.lr'
```
even though `en` is the primary alternative and there is no `contents+en.lr`. So something needs to fixing.

My guess by looking at the chart on the docs page would be that `this.contents` would return the contents of file in the column in the far right of that chart, and should only error when the file used is truly missing. If it returned the proper content object (and it won't always right now), then you could

1) query and parse `this.contents.filename`. This is undocumented AFIAK, but [currently in the code](https://github.com/lektor/lektor/blob/master/lektor/filecontents.py#L11) This returns a string that looks like `/home/.../content/contents+en.lr`

2) query for an attribute that doesn't yet exist that does the parsing for you, and just returns the same data, i.e., what is inside `contents[??].lr`

**In Summary**
This requires a bugfix related to handling the primary alt. On top of that you can do some string parsing, or we can add an attribute to the contents object. The enhancement would be pretty easy to do, and seems like a general thing anyone might want to know, so I'm for adding it.

Sound like a plan @goanpeca?

**Comment by goanpeca** [comment]
@nixjdm I am bit confused on the example you show. I basically want to know if a given alternative file REALLY exists or not, so either `this.contents` returns `None`, or we add some method/property like `this.file_exists` or something.

By the way in the meantime I solved the issues with a custom plugin :-\

**Comment by nixjdm** [comment]
@goanpeca That's fine that you found another way to get what you need. :)

This still sounds like a something that should change though. Let me try to explain in another way.

The only time `this.contents` should error is when you are rendering it on a template where neither that language `contents+<alt>.lr` nor `contents.lr` exists. For example, if you render this on the `fr` alt while only `contents+en.lr` exists, and no `contents.lr` does. That would bring it in line with [this chart](https://www.getlektor.com/docs/content/alts/#what-s-loaded). Currently this has issues and I'll call it a bug.

If that were working as described, you could determine if the `fr` alternative was made yet by checking `this.contents.filename` for the existence of the substring `fr`. E.g. 
```
{% if '+fr' in this.contents.filename %}
```

Clear as mud?

**Comment by goanpeca** [comment]
Ok, when is 

>The only time this.contents should error is when you are rendering it on a template where neither that language contents+<alt>.lr nor contents.lr exists.

ever going to happen ??

**Comment by nixjdm** [comment]
That will happen upon misconfiguration. And then it *should* error, right? In your situation, you shouldn't be getting errors because presumably you actually have a `contents.lr` or the proper `contents+<alt>.lr`.

**Comment by nixjdm** [comment]
Here we go. @goanpeca just use this, it's in Lektor now and pretty condensed.
```
{% if 'fr' in get_alts() %}
  alt exists
{% endif %}
```
https://www.getlektor.com/docs/api/db/get-alts/

**Comment by goanpeca** [comment]
Ha... :-p thanks ;-)

**Comment by goanpeca** [comment]
Then we should close this I guess?

**Comment by nixjdm** [comment]
I think so. I've convinced myself filecontents.py doesn't need to change because of any of this. I got it :)
